spec/acceptance/rest/message_spec.rb

```diff
@@ -91,6 +91,109 @@
       end
     end
 
+    context 'idempotency' do
+      let(:id) { random_str }
+      let(:name) { 'event' }
+      let(:data) { random_str }
+
+      context 'when ID is not included' do
+        context 'with Message object' do
+          let(:message) { Ably::Models::Message.new(data: data) }
+
+          it 'publishes the same message three times' do
+            3.times { channel.publish [message] }
+            expect(channel.history.items.length).to eql(3)
+          end
+        end
+
+        context 'with #publish arguments only' do
+          it 'publishes the same message three times' do
+            3.times { channel.publish 'event', data }
+            expect(channel.history.items.length).to eql(3)
+          end
+        end
+      end
+
+      context 'when ID is included' do
+        context 'with Message object' do
+          let(:message) { Ably::Models::Message.new(id: id, data: data) }
+
+          it 'three REST publishes result in only one message being published' do
+            3.times { channel.publish [message] }
+            expect(channel.history.items.length).to eql(1)
```

Comment:
Why not testing also the message id here?

Reply:
Ok, fixing now.